presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -521,12 +517,17 @@ public PlanOptimizers(
                 statsCalculator,
                 costCalculator,
                 new TranslateExpressions(metadata, sqlParser).rules()));
+        // After this point, all planNodes should not contain OriginalExpression
 
         // TODO: Do not move other PlanNode to SPI until this rule is moved to the end of logical planning (i.e., where AddExchanges lives)
         // TODO: The connector optimizer should not change plan shape (computations) at this point util #12960 is landed. (e.g., connector may pushdown filters to table scan but cannot add a new node)
```

Comment:
I added this line because it makes more obvious where the RowExpression translation begins and where to put the moved optimization rules. Otherwise, the following optimization rule will seem mixed with TranslateExpression and less obvious.